test/e2e/network_test.go

```diff
@@ -301,12 +302,25 @@ func (tc *testContext) testEastWestNetworking(t *testing.T) {
 
 	linuxServerDeployment, err := tc.deployLinuxWebServer()
 	require.NoError(t, err)
-	defer tc.deleteDeployment(linuxServerDeployment.GetName())
 	linuxServerClusterIP, err := tc.createService(linuxServerDeployment.GetName(), 8080, v1.ServiceTypeClusterIP,
 		*linuxServerDeployment.Spec.Selector)
 	require.NoError(t, err)
-	defer tc.deleteService(linuxServerClusterIP.GetName())
-	linuxServerIP, err := tc.getPodIP(*linuxServerDeployment.Spec.Selector)
+	// best effort to collect Linux web server logs before deleting the deployment and service
+	defer func(tc *testContext, deployment *appsv1.Deployment) {
+		err := tc.collectDeploymentLogs(deployment)
+		if err != nil {
+			log.Printf("error collecting deployment logs: %v", err)
+		}
+		err = tc.deleteDeployment(linuxServerDeployment.GetName())
```

Comment:
This should be consistent, either deployment or linuxServerDeployment

Reply:
A helper to delete and collect logs could be made if this pattern is repeating, but if its only twice it is fine without

Reply:
PTAL